### PR TITLE
ociodisplay gets an inverse control

### LINKED ISSUE #3648: [FEATURE REQUEST] Add an inverse modifier to oiiotool's --ociodisplay option
**Problem Description**

The oiiotool --ociodisplay option is very useful for applying an OCIO DisplayViewTransform.  This takes an image in a "src" color space and then applies a "view" transform for a specific "display".  For example, one might take a source image in ACEScg space and apply the "ACES 1.0 - SDR Video" view for the "sRGB - Display", using either of the following:
 
```
% oiiotool acescg.exr --ociodisplay:from="ACEScg" "sRGB - Display" "ACES 1.0 - SDR Video" -o srgb.jpg
% oiiotool acescg.exr --iscolorspace "ACEScg" --ociodisplay "sRGB - Display" "ACES 1.0 - SDR Video" -o srgb.jpg
```
However, it is often useful to be able to apply the DisplayViewTransform in the inverse direction.  For example, to convert sRGB back to ACEScg, inverting a specific view transform.  The ocioconvert tool (one of the OCIO command-line tools) is already able to do this, as follows:

```
% ocioconvert --invertview srgb.exr "sRGB - Display" "ACES 1.0 - SDR Video" acescg.exr "ACEScg"
```
Inverting a DisplayViewTransform is easily possible via the OCIO v2 API.  Simply set the direction of the DisplayViewTransform to TRANSFORM_DIR_INVERSE.  

**Proposed solution**

The --ociofiletransform option in oiiotool already has an "inverse" modifier.  So the proposed solution is to add an "inverse" modifier to --ociodisplay.

### PR DESCRIPTION
**Pull request by lgritz**
Fixes #3648

### CODE REVIEW

File: src/libOpenImageIO/color_ocio.cpp [hunk 1]
```diff
@@ -1848,7 +1876,7 @@ bool
 ImageBufAlgo::ociodisplay(ImageBuf& dst, const ImageBuf& src,
                           string_view display, string_view view,
                           string_view from, string_view looks, bool unpremult,
-                          string_view key, string_view value,
+                          bool inverse, string_view key, string_view value,
                           ColorConfig* colorconfig, ROI roi, int nthreads)
 {
```
Comment (doug-walker) [inline src/libOpenImageIO/color_ocio.cpp:1881]: Don't know how to select the line 1872, but don't you want to pass inverse there too?

File: src/libOpenImageIO/color_ocio.cpp [hunk 2]
```diff
@@ -1889,19 +1917,47 @@ ImageBufAlgo::ociodisplay(ImageBuf& dst, const ImageBuf& src,
...
-                          bool unpremult, string_view key, string_view value,
-                          ColorConfig* colorconfig, ROI roi, int nthreads)
+                          bool unpremult, bool inverse, string_view key,
+                          string_view value, ColorConfig* colorconfig, ROI roi,
+                          int nthreads)
 {
     ImageBuf result;
     bool ok = ociodisplay(result, src, display, view, from, looks, unpremult,
-                          key, value, colorconfig, roi, nthreads);
+                          unpremult, key, value, colorconfig, roi, nthreads);
```
Comment (doug-walker) [inline src/libOpenImageIO/color_ocio.cpp:1926]: unpremult --> inverse

**Review by doug-walker — CHANGES_REQUESTED** [review]
Thanks for adding this Larry!

File: src/libOpenImageIO/color_ocio.cpp (hunk 1 above)
Comment (lgritz) [reply src/libOpenImageIO/color_ocio.cpp:1881]: yes, I think you're right

File: src/libOpenImageIO/color_ocio.cpp (hunk 2 above)
Comment (lgritz) [reply src/libOpenImageIO/color_ocio.cpp:1926]: ouch, yes

**Comment by lgritz** [comment]
Updated the PRs to address @doug-walker's suggestions.

**Review by doug-walker — APPROVED** [review]
LGTM!

**Comment by lgritz** [comment]
Because this isn't *tested*, I'm going to hold off on a merge while I do a little tinkering...

I'm exploring more direct support for OpenColorIO 2.2 and its built-in configs, which will allow me to test display transforms (merely because it has some, unlike the sample config that's in the testsuite now).

If minimal support of this is easy to achieve, I'll add it to this PR (even if the full range of tests only work when we're building against OCIO 2.2). But if that looks like a bigger job that doesn't make sense to append here, I'll merge this PR first (despite not being tested) and the follow up with a separate one with 2.2 support and more testing.

**Comment by lgritz** [comment]
I'm going to merge this now, because the follow-on has grown to be more complex.

**Pull request merged**